src/OpenLoco/src/GameCommands/Docks/RemovePort.cpp

```diff
@@ -0,0 +1,151 @@
+#include "RemovePort.h"
+#include "Economy/Economy.h"
+#include "Localisation/StringIds.h"
+#include "Map/StationElement.h"
+#include "Map/SurfaceElement.h"
+#include "Map/TileElement.h"
+#include "Map/TileManager.h"
+#include "Map/Track/TrackData.h"
+#include "Objects/DockObject.h"
+#include "Objects/ObjectManager.h"
+#include "ViewportManager.h"
+#include "World/Station.h"
+#include "World/StationManager.h"
+
+namespace OpenLoco::GameCommands
+{
+    // TODO: Copied from Industry.h
+    struct Unk4F9274
+    {
+        World::Pos2 pos;
+        uint8_t index;
+    };
+
+    // TODO: Copied from Industry.cpp
+    static const std::array<Unk4F9274, 4> word_4F927C = {
+        Unk4F9274{ { 0, 0 }, 0 },
+        Unk4F9274{ { 0, 32 }, 1 },
+        Unk4F9274{ { 32, 32 }, 2 },
+        Unk4F9274{ { 32, 0 }, 3 },
+    };
+
+    static World::StationElement* getStationEl(const World::Pos3& pos)
+    {
+        auto tile = World::TileManager::get(pos);
+        for (auto& el : tile)
+        {
+            auto* stationEl = el.as<World::StationElement>();
+            if (stationEl == nullptr)
+            {
+                continue;
+            }
+
+            if (stationEl->baseHeight() == pos.z)
+            {
+                return stationEl;
+            }
+        }
+
+        return nullptr;
+    }
+
+    // 0x00494706
+    static bool removePortTileElements(const World::Pos3& pos, const uint8_t flags)
+    {
+        for (auto& searchTile : word_4F927C)
+        {
+            const auto portPos = World::Pos3(searchTile.pos + pos, pos.z);
+
+            if ((flags & (Flags::aiAllocated | Flags::apply)) != 0)
+            {
+                auto tile = World::TileManager::get(portPos);
+                auto* surfaceEl = tile.surface();
+                if (surfaceEl != nullptr)
+                {
+                    surfaceEl->setAiAllocated(false);
+                }
+            }
+
+            if ((flags & Flags::apply) == 0)
+            {
+                continue;
+            }
+
+            auto* stationEl = getStationEl(portPos);
+            if (stationEl == nullptr)
```

Comment:
I know this matches vanilla but we should really move this if before the if ((flags & Flags::apply) == 0) otherwise you could partially remove a dock!